### PR TITLE
Feat(eos_cli_config_gen): Add OSPF timers to router ospf

### LINKED ISSUE #1067: Add OSPF timers in Cli config gen
**Issue Type**

- [ ] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**
No.

There is no structured data in the router ospf configuration on the cli config gen for timers: throttle, lsa, spf, etc

**Describe the solution you'd like**

To have a key in the router_ospf section for timers

**Describe alternatives you've considered**

Add keys for the timers

Something like:

```yaml
router_ospf:
 '2':
   timers:
      throttle:
          sfp: <string as value>
          lsa: <string as value>
      lsa:
         rx: <string as value>
         tx: <string as value>
```

**Additional context**

### PR DESCRIPTION
**Pull request by titom73**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #1067 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Extend data model to support OSPF timers:

```yaml
router_ospf:
  process_ids:
    <process-id>:
      timers:
        lsa:
          rx_min_interval: < 0-600000 - Min interval in msecs between accepting the same LSA >
          tx_delay:
            initial: < 0-600000 - Delay to generate first occurrence of LSA in msecs >
            min: < 1-600000 Min delay between originating the same LSA in msecs >
            max: < 1-600000 Maximum delay between originating the same LSA in msecs >
        spf_delay:
          initial: < 0-600000 - Initial SPF schedule delay in msecs >
          min: < 0-65535000  Min Hold time between two SPFs in msecs >
          max: < 0-65535000  Max wait time between two SPFs in msecs >
```

Status:

- [x] Config rendering
- [ ] Documentation rendering

## How to test

Run molecule

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 1]
```diff
@@ -2114,6 +2114,19 @@ router_ospf:
         - < interface_1 >
         - < interface_2 >
       max_lsa: < integer >
+      timers:
+        lsa:
+          rx: < 0-600000 - Min interval in msecs between accepting the same LSA >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: this should be `rx_min_interval`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 2]
```diff
@@ -2114,6 +2114,19 @@ router_ospf:
...
+          tx:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Maybe flatten by using `tx_delay`?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 3]
```diff
@@ -2114,6 +2114,19 @@ router_ospf:
...
+            delay:
+              initial: < 0-600000 - Delay to generate first occurrence of LSA in msecs >
+              min: < 1-600000 Min delay between originating the same LSA in msecs >
+              max: < 1-600000 Maximum delay between originating the same LSA in msecs >
+        spf:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Maybe flatten by using `spf_delay`?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 [hunk 4]
```diff
@@ -120,6 +120,44 @@
...
+{%     for process in router_ospf.process_ids %}
+{%         if router_ospf.process_ids[process].timers is arista.avd.defined %}
+{%             set has.found = true %}
+{%         endif %}
+{%     endfor %}
+{%     if has.found is arista.avd.defined(true) %}
+
+### Router OSPF timers
+
+| Process ID | LSA rx | LSA tx | SPF |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: If we keep the multiple timer values in the same field, I would include some assistance to the reader like `LSA tx (initial/min/max)` and use `/` as delimiter.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 [hunk 5]
```diff
@@ -120,6 +120,44 @@
...
+| ---------- | ------ | ------ | --- |
+{%         for process in router_ospf.process_ids | arista.avd.natural_sort %}
+{%             if router_ospf.process_ids[process].timers is arista.avd.defined %}
+{%                 if router_ospf.process_ids[process].timers.lsa.rx is defined %}
+{%                     set lsa_rx = router_ospf.process_ids[process].timers.lsa.rx %}
+{%                 else %}
+{%                     set lsa_rx = 'unset' %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: to be consistent with most newer avd docs we should just use `-`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 [hunk 6]
```diff
@@ -120,6 +120,44 @@
...
+{%                 endif %}
+{%                 if router_ospf.process_ids[process].timers.lsa.tx is defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: this should test for all the child values with `and`. Also use `arista.avd.defined`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 [hunk 7]
```diff
@@ -120,6 +120,44 @@
...
+{%     endfor %}
+{%     if has.found is arista.avd.defined(true) %}
+
+### Router OSPF timers
+
+| Process ID | LSA rx | LSA tx | SPF |
+| ---------- | ------ | ------ | --- |
+{%         for process in router_ospf.process_ids | arista.avd.natural_sort %}
+{%             if router_ospf.process_ids[process].timers is arista.avd.defined %}
+{%                 if router_ospf.process_ids[process].timers.lsa.rx is defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: use `arista.avd.defined` or just replace with `set lsa_rx = router_ospf.process_ids[process].timers.lsa.rx | arista.avd.default('-')`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 [hunk 8]
```diff
@@ -29,6 +29,35 @@ router ospf {{ process_id }}
 {%     if router_ospf.process_ids[process_id].max_lsa is arista.avd.defined %}
    max-lsa {{ router_ospf.process_ids[process_id].max_lsa }}
 {%     endif %}
+{%     if router_ospf.process_ids[process_id].timers.lsa.rx is arista.avd.defined %}
+   timers lsa rx min interval {{ router_ospf.process_ids[process_id].timers.lsa.rx }}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].timers.lsa.tx.delay is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2]: this should check for all the child values at once, so we don't generate config with only some of the values. same for spf.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 1 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Fixed in commit e4265b9a

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 3 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Flatten `spf` and `tx` to `spf_delay` and `tx_delay` with commit e4265b9a

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 (hunk 4 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: Updated with e4265b9a

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 (hunk 6 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: Fixed with e4265b9a

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 (hunk 8 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2]: Fixed with commit e4265b9a

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 (hunk 7 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: Fixed with commit  ad47c327

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 (hunk 5 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: Fixed with commit  ad47c327

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 2 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Flatten `spf` and `tx` to `spf_delay` and `tx_delay` with commit e4265b9

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 [hunk 9]
```diff
@@ -120,6 +120,44 @@
 {%             endif %}
 {%         endfor %}
 {%     endif %}
+{# OSPF Timers #}
+{%     set has.found = false %}
+{%     for process in router_ospf.process_ids %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: missed this last time. Please add `| arista.avd.natural_sort` to this `for` so we catch if not defined.

**Review by ClausHolbechArista — COMMENTED** [review]
Looking good. A minor update and we should be good.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM!

**Pull request merged**